nixos/modules/services/video/epgstation.nix

```diff
@@ -0,0 +1,335 @@
+      default = { appenders = [ "stdout" ]; level = "info"; };
+      system = { appenders = [ "stdout" ]; level = "info"; };
+      access = { appenders = [ "stdout" ]; level = "info"; };
+      stream = { appenders = [ "stdout" ]; level = "info"; };
+    };
+  };
+in
+{
+  options.services.epgstation = {
+    enable = mkEnableOption pkgs.epgstation.meta.description;
+
+    usePreconfiguredStreaming = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Use preconfigured streaming options. It's based on upstream defaults.
+
+        Upstream defaults:
+        <link xlink:href="https://github.com/l3tnun/EPGStation/blob/master/config/config.sample.json"/>
+      '';
+    };
+
+    readOnlyOnce = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Don't reload configuration files at runtime.";
+    };
+
+    serverPort = mkOption {
+      type = types.port;
+      default = 20772;
+      description = ''
+        Port for EPGStation to listen on. Be careful because EPGStation listens
+        on a total of two ports: <literal>serverPort</literal> for regular HTTP
+        and <literal>serverPort + 1</literal> for websockets.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Open ports in the firewall for the EPGStation web interface.
+
+        <warning>
+          <para>
+            It is generally advised against exposing EPGStation to the open
+            internet. Only use it inside a trusted local network, or consider
+            putting it behind a VPN if you want remote access.
+          </para>
+        </warning>
+      '';
+    };
+
+    basicAuth = {
+      user = mkOption {
+        type = with types; nullOr str;
+        default = null;
+        example = "epgstation";
+        description = ''
+          Basic auth username for EPGStation. If <literal>null</literal>, basic
+          auth will be disabled.
+
+          <warning>
+            <para>
+              Basic authentication has known weaknesses, the most critical being
+              that it sends passwords over the network in clear text. Use this
+              feature to control access to EPGStation within your family and
+              friends, but dont't rely on it for security.
+            </para>
+          </warning>
+        '';
+      };
+
+      password = mkOption {
+        type = types.str;
+        default = "INSECURE_GO_CHECK_CONFIGURATION_NIX";
+        example = "password";
+        description = ''
+          Basic auth password for EPGStation. The password doesn't need to be
+          set if either basic auth is disabled or
+          <option>basicAuth.passwordFile</option> has been set.
+
+          <warning>
+            <para>
+              This stores the plaintext password in the world-readable Nix
+              store.  Consider using <option>basicAuth.passwordFile</option>
```

Comment:
How about just not providing such a `password` option. Users can still explicitly put the password in the store with `passwordFile = pkgs.writeText "pw" "..."`. Same for the other password option

Reply:
Fixed.